### PR TITLE
APIGOV-17331 - Add instruction for Azure traceability agent

### PR DESCRIPTION
**Pull request by sbolosan**
Automatically generated by Netlify CMS

### CODE REVIEW

**Comment by sbolosan** [comment]
@cbordier @lbadenhop can you guys review the structure change before I add any "meat" to the traceability section?  I divided this page into discovery and traceability now... they have their own instructions...

**Comment by sbolosan** [comment]
Documentation story -> https://jira.axway.com/browse/APIGOV-17331

**Comment by sbolosan** [comment]
@cbordier good catch on the env var.
Aside from that, are you ok with the breakdown structure?  if you're ok, I can add my content.

**Comment by lbadenhop** [comment]
@sbolosan I tried to delete your comment that contains the Jira link, but I couldn't. Please delete the link -- not allowed to include links to Jira.

**Comment by sbolosan** [comment]
@lbadenhop sorry.  did not know.  Deleted.

**Comment by cbordier** [comment]
@sbolosan, can you review and complete?

**Comment by sbolosan** [comment]
@cbordier I will add the remaining content and commit for review

**Review by sbolosan — COMMENTED** [review]
Looks good for me.  Only comment i have is maybe, since we don't have any images on event hub/diag settings, azure monitor, etc, we add some links?

**Review by sbolosan — COMMENTED** [review]
@cbordier @lbadenhop approved!

**Review by cbordier — APPROVED** [review]
I approve the changes.
@lbadenhop, please proceed

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 1]
```diff
@@ -50,23 +50,29 @@ AMPLIFY Central supports two regions, US (default) and EU. The data (APIs, traff
...
+
+## Connect Azure API Management services to AMPLIFY Central for discovery
 
 The following is a high-level overview of the required steps to connect Azure API Management services to AMPLIFY Central:
 
 * Create an Azure Service principal
 * Create a service account for the agent to communicate with the AMPLIFY platform
 * Create an environment to group the APIs
-* Pull the agent from Docker
+* Pull the discovery agent from Docker
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:75]:
```suggestion
* Pull the Discovery Agent from Docker
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 2]
```diff
@@ -50,23 +50,29 @@ AMPLIFY Central supports two regions, US (default) and EU. The data (APIs, traff
...
 * Update the Azure agent YAML file or use an environment variable file to configure the access to Azure.
 
 ### Creating Azure Service principal
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:78]:
```suggestion
### Creating an Azure service principal
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 3]
```diff
@@ -101,15 +107,9 @@ Notes:
 
 You can retrieve your subscription id with the command: `az account show --query id`
 
-### Prepare AMPLIFY Central
+### Prepare discovery agent configuration
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:110]:
```suggestion
### Prepare Discovery Agent configuration
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 4]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
-Second, run the following docker command: `docker run --env-file <PATH>/env_vars.env -v <PATH>/<SA-Key-folder>:/keys beano.swf-artifactory.lab.phx.axway.int/beano/azure-discovery-agent`.
+Second, run the following docker command: `docker run --env-file <PATH>/da_env_vars.env -v <PATH>/<SA-Key-folder>:/keys beano.swf-artifactory.lab.phx.axway.int/beano/azure-discovery-agent`.
 
 Note: `SA-Key-folder` is the folder name that contains the service account keys (private_key.pem / public_key.pem).
 
+## Connect Azure API Management services to AMPLIFY Central for traceability
+
+The following is a high-level overview of the required steps to connect Azure API Management services to AMPLIFY Central:
+
+* Create an Azure Event Hub Namespace and Event Hub
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:160]:
```suggestion
* Create an Azure Event Hubs namespace and event hub
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 5]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+* Create an Azure Diagnostic Setting
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:161]:
```suggestion
* Create an Azure diagnostic setting
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 6]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+* Configure Azure API's for Azure monitoring
+* Pull the traceability agent from Docker
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:163]:
```suggestion
* Pull the Traceability Agent from Docker
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 7]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+* Update the Azure agent YAML file or use an environment variable file to configure the access to Azure.
+
+### Creating Azure Event Hub Namespace and Event Hub
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:166]:
```suggestion
### Creating Azure Event Hubs namespace and event hub
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 8]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+Azure Event Hubs is a big data streaming platform and event ingestion service. Refer to [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about)
+
+In order to create an Azure event hub, it is first required to create an event hub namespace. For that, go to Azure Portal and select the \[Event Hubs\] service. Then create the event hub nanmspace. Select the appropriate Azure subscription and resource group where this namespace will be available. Be sure to create the event hub namespace in the same region as your API Management service. For the pricing, we recommend the Standard one. When creating the event hb namespace, Azure will automatically attach to it an access policy. We recommend to keep the default values provided.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:170]:
```suggestion
In order to create an Azure event hub, it is first required to create an Event Hubs namespace. For that, go to Azure Portal and select the \[Event Hubs\] service. Then create the Event Hubs namespace. Select the appropriate Azure subscription and resource group where this namespace will be available. Be sure to create the Event Hubs namespace in the same region as your API Management service. For the pricing, we recommend the Standard one. When creating the Event Hubs namespace, Azure will automatically attach to it an access policy. We recommend to keep the default values provided.
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 9]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+* Create an Azure Event Hub Namespace and Event Hub
+* Create an Azure Diagnostic Setting
+* Configure Azure API's for Azure monitoring
+* Pull the traceability agent from Docker
+* Update the Azure agent YAML file or use an environment variable file to configure the access to Azure.
+
+### Creating Azure Event Hub Namespace and Event Hub
+
+Azure Event Hubs is a big data streaming platform and event ingestion service. Refer to [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about)
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:168]:
```suggestion
Azure Event Hubs is a big data streaming platform and event ingestion service. Refer to [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about).
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 10]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+In order to create an Azure event hub, it is first required to create an event hub namespace. For that, go to Azure Portal and select the \[Event Hubs\] service. Then create the event hub nanmspace. Select the appropriate Azure subscription and resource group where this namespace will be available. Be sure to create the event hub namespace in the same region as your API Management service. For the pricing, we recommend the Standard one. When creating the event hb namespace, Azure will automatically attach to it an access policy. We recommend to keep the default values provided.
+
+Once the namespace is created, you can add an event Hub to this namespace.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:172]:
```suggestion
Once the namespace is created, you can add an event hub to this namespace.
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 11]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+What you need to remember from this event hub creation:
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:174]: Remember the following when creating the event hub, as you'll need the information for the Traceability Agent configuration:

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 12]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+* the **event hub namespace name**
+* the **event hub name**
+* the **policy name** attached to the event hub namespace (RootManageSharedAccessKey by default)
+* the **policy primary or second key**
+
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:180]: Remove "the" from each of the above bullets

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 13]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+Azure Event Hubs is a big data streaming platform and event ingestion service. Refer to [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about)
+
+In order to create an Azure event hub, it is first required to create an event hub namespace. For that, go to Azure Portal and select the \[Event Hubs\] service. Then create the event hub nanmspace. Select the appropriate Azure subscription and resource group where this namespace will be available. Be sure to create the event hub namespace in the same region as your API Management service. For the pricing, we recommend the Standard one. When creating the event hb namespace, Azure will automatically attach to it an access policy. We recommend to keep the default values provided.
+
+Once the namespace is created, you can add an event Hub to this namespace.
+
+What you need to remember from this event hub creation:
+
+* the **event hub namespace name**
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:176]:
```suggestion
* the **Event Hubs namespace name**
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 14]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+* the **event hub name**
+* the **policy name** attached to the event hub namespace (RootManageSharedAccessKey by default)
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:178]:
```suggestion
* the **policy name** attached to the Event Hubs namespace (RootManageSharedAccessKey by default)
```

File: content/en/docs/central/connect-azure-gateway/_index.md (hunk 4 above)
Comment (cbordier) [reply content/en/docs/central/connect-azure-gateway/_index.md:160]: Event Hub is an Azure service and Microsoft spell it with upercase letter for each words.

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 15]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+* the **policy primary or second key**
+
+Traceability Agent will need these information in its configuration.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:181]: Remove this sentence. Reword line #174.

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 16]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+### Creating Azure Diagnostic Settings
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:183]:
```suggestion
### Creating Azure diagnostic settings
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 17]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+The Azure Diagnostic settings will link the Resource group to the Event Hub.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:185]:
```suggestion
The Azure diagnostic settings will link the Resource group to the Event Hub.
```

File: content/en/docs/central/connect-azure-gateway/_index.md (hunk 11 above)
Comment (cbordier) [reply content/en/docs/central/connect-azure-gateway/_index.md:174]: Updated

File: content/en/docs/central/connect-azure-gateway/_index.md (hunk 12 above)
Comment (cbordier) [reply content/en/docs/central/connect-azure-gateway/_index.md:180]: updated

File: content/en/docs/central/connect-azure-gateway/_index.md (hunk 10 above)
Comment (cbordier) [reply content/en/docs/central/connect-azure-gateway/_index.md:172]: Event Hub instead

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 18]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+For creating it, go to Azure portal, open your resource group and open the Monitoring / Diagnostic settings page. Simply Add a diagnostic setting from there. Give it a name and select "GatewayLogs" and "Stream to an event hub" options. The "Stream to event hub option" requires the previous information from event hub namespace and event hub name. Save your changes.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:187]:
```suggestion
To create a diagnostic setting: 
1. Go to Azure portal, open your resource group and open the Monitoring / Diagnostic settings page. 
2. Add a diagnostic setting and give it a name. 
3. Select the **GatewayLogs** and **Stream to an event hub** options. The **Stream to event hub** option requires the previous information from Event Hubs namespace and event hub name. 
4. Save your changes.
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 19]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+Now Azure is ready to register your APi traffic to the event hub.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:192]:
```suggestion
Azure is now ready to register your API traffic to the event hub.
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 20]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+### Configuring API Azure Monitoring
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:194]:
```suggestion
### Configure API Azure monitoring
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 21]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+By default Azure does not monior any API traffic nor their headers. You have to explicitly ask for that. You can do it per API or for all APIs present in the API Managment service.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:196]:
```suggestion
By default, Azure does not monitor API traffic or their headers. You have to explicitly ask for that. You can do it per API or for all APIs present in the API Management service.
```

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 22]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+For this last step, open the settings of an API (or All APIs) and select the Azure Monitor tab. Enable it, select the Vebose mode and add all headers you want to track. These headers can belongs to any request/response (frontend/backend) or you can decide to have different header to log using the advanced options.
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:198]:
```suggestion
To configure monitoring: 
1. Open the settings of an API (or all APIs). 
2. Select the Azure Monitor tab. 
3. Enable it. 
4. Select the Verbose mode and add all headers you want to track. These headers can belongs to any request/response (frontend/backend) or you can choose different headers to log using the advanced options.
```

Comment (lbadenhop) [reply content/en/docs/central/connect-azure-gateway/_index.md:198]: I don't understand Step 3, "Enable it."

File: content/en/docs/central/connect-azure-gateway/_index.md [hunk 23]
```diff
@@ -149,14 +149,92 @@ For more information about the agent variables, refer to [Reference - Agent conf
...
+
+### Prepare traceability agent configuration
```
Comment (lbadenhop) [inline content/en/docs/central/connect-azure-gateway/_index.md:200]:
```suggestion
### Prepare Traceability Agent configuration
```

**Review by lbadenhop — COMMENTED** [review]
I wasn't sure of some of the edits, so I'm leaving it up to you to accept each edit.

File: content/en/docs/central/connect-azure-gateway/_index.md (hunk 22 above)
Comment (cbordier) [reply content/en/docs/central/connect-azure-gateway/_index.md:198]: there is a check box to enable the Monitoring. If uncheck, no monitor.

**Review by lbadenhop — APPROVED** [review]

**Pull request merged**